server/jetstream_api.go

```diff
@@ -980,14 +985,29 @@ func (s *Server) jsStreamLookupRequest(sub *subscription, c *client, subject, re
 		s.sendAPIResponse(c, subject, reply, string(msg), s.jsonResponse(&resp))
 		return
 	}
+
 	if isEmptyRequest(msg) {
 		resp.Error = &ApiError{Code: 400, Description: "subject required"}
 		s.sendAPIResponse(c, subject, reply, string(msg), s.jsonResponse(&resp))
 		return
 	}
+
 	subj := string(msg)
+
+	if bytes.HasPrefix(msg, []byte("{")) {
```

Comment:
it's all a bit fragile since `{"subject":"x"}` is a valid subject via`IsValidSubject` and `HasPrefix(...."{")` isnt exactly robust, could just always try to first parse as JSON I guess and check if req.Subject is empty at a bit of a cost for bare word requests.

Reply:
We want to avoid json if we can, very expensive in its current form unfortunately, so anything in fast path needs to be scrutinized.

Reply:
Yes, so thats what this PR does, use it only when it seems likely that its JSON.

Reply:
Take a look at `nextReqFromMsg()` in consumer.go

Reply:
only real difference there is `req[0] == '{'` we cant convert these 3 if statements into a switch like there, shall I use that instead of HasPrefix?

Reply:
That line works because of Trim above.

Reply:
added trim

Reply:
ok, why not just check msg[0] then for '{' vs calling out to HasPrefix? Just curious.

Reply:
no real reason, can do if you prefer, it's still fragile though since "{" is a valid stream name thats more what I was on about earlier saying this check is inherintly not robust.

Reply:
If parsing fails you can fall back after detecting a '{'

Reply:
add that